docs/source/code-generation.rst

```diff
@@ -18,8 +18,8 @@ To use Palm code generation you will need:
    project's .palm directory.
 2. A YAML configuration file called ``template-config.yaml`` - this is a
    configuration file that describes how to generate code.
-3. A palm command which is decorated with the ``@click.pass_context`` decorator and
-   calls ``ctx.obj.generate(template_path, output_path, replacements)``.
+3. A palm command which is decorated with the ``@click.pass_obj`` decorator and
+   calls ``environment.generate(template_path, output_path, replacements)``.
```

Comment:
This change to docs also needs to be reverted

Reply:
Why is this necessary? The code generation functions on `ctx.obj` which is equivalent to `environment`. If we change this documentation then it won't reflect any of the commands that this PR updates to use `pass_obj` and `environment` over `ctx.obj`.

This `generate` function is not the Containerizer's `generate` function, it's the one for the `Environment` class. We reverted my change to Containerizer because it would have prevented other things that use `ctx` from working, but `Environment` isn't reverted.

Reply:
yup, you're right - fine to keep this as-is